### PR TITLE
<SectionHelper /> - Close Button

### PR DESCRIPTION
**Pull request by liorzisman**

### CODE REVIEW

File: src/SectionHelper/CloseButton/CloseButton.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import Button from '../../Button';
+import Close from '../../Icons/dist/components/Close';
+
+const CloseButton = properties => (
+  <Button dataHook="close" height="medium" theme="close-transparent" onClick={properties.onClick || (e => e.preventDefault())}>
```
Comment (argshook) [inline src/SectionHelper/CloseButton/CloseButton.js]: i wonder if you really benefit splitting this component to separate file. But anyhow:

* please split long lines
* add `type="button`, then no `preventDefault`
* destructure `properties` and add `propTypes`

File: src/SectionHelper/CloseButton/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export {default} from './CloseButton.js';
```
Comment (argshook) [inline src/SectionHelper/CloseButton/index.js]: you know, i usually am quiet about this but such files are reaaally not necessary :)

File: src/SectionHelper/SectionHelper.driver.js [hunk 3]
```diff
@@ -1,9 +1,12 @@
+import ReactTestUtils from 'react-dom/test-utils';
+
 const sectionHelperDriverFactory = ({element}) => {
   const classExists = className => element.classList.contains(className);
 
   return {
     exists: () => !!element,
     titleText: () => element.querySelector('[data-hook="title"]').textContent,
+    clickClose: () => ReactTestUtils.Simulate.click(element.querySelector('[data-hook="close"]')),
```
Comment (argshook) [inline src/SectionHelper/SectionHelper.driver.js]: just `close`?

File: src/SectionHelper/SectionHelper.js [hunk 4]
```diff
@@ -35,12 +37,16 @@ SectionHelper.propTypes = {
...
+  canBeClosed: PropTypes.bool,
+  onClose: PropTypes.func
 };
 
 SectionHelper.defaultProps = {
   appearance: 'warning',
-  title: null
+  title: null,
+  canBeClosed: false,
+  onClose: e => e.preventDefault()
```
Comment (argshook) [inline src/SectionHelper/SectionHelper.js]: perhaps just `onClose` is enough? if set, show button, else don't show. Then you don't need `canBeClosed`

File: src/SectionHelper/SectionHelper.js [hunk 5]
```diff
@@ -35,12 +37,16 @@ SectionHelper.propTypes = {
   /** Sets the look and feel of the component */
   appearance: PropTypes.oneOf(Object.keys(HELPER_APPEARANCE)),
   /** Adds text as the title */
-  title: PropTypes.node
+  title: PropTypes.node,
+  canBeClosed: PropTypes.bool,
+  onClose: PropTypes.func
```
Comment (shlomitc) [inline src/SectionHelper/SectionHelper.js]: don't forget to add function documentation

File: src/SectionHelper/SectionHelper.js (hunk 4 above)
Comment (liorzisman) [reply src/SectionHelper/SectionHelper.js]: considered that, im glad there's another 'vote' for it. will apply this logic

File: src/SectionHelper/SectionHelper.driver.js (hunk 3 above)
Comment (liorzisman) [reply src/SectionHelper/SectionHelper.driver.js]: probably can be more descriptive indeed

File: src/SectionHelper/CloseButton/index.js (hunk 2 above)
Comment (liorzisman) [reply src/SectionHelper/CloseButton/index.js]: wanted to make the import nicer. will just move everything to the index

File: src/SectionHelper/CloseButton/CloseButton.js (hunk 1 above)
Comment (liorzisman) [reply src/SectionHelper/CloseButton/CloseButton.js]: Why not allow `SectionHelper` refer this as a separate unit? I think its beneficial indeed
Will apply the other modifications

File: src/SectionHelper/SectionHelper.js (hunk 5 above)
Comment (liorzisman) [reply src/SectionHelper/SectionHelper.js]: done

File: src/SectionHelper/SectionHelper.js [hunk 6]
```diff
@@ -35,12 +37,15 @@ SectionHelper.propTypes = {
...
+  title: PropTypes.node,
+  /** When provided, will make a close button appear and invoke it upon click */
+  onClose: PropTypes.func
 };
 
 SectionHelper.defaultProps = {
   appearance: 'warning',
-  title: null
+  title: null,
+  onClose: e => e.preventDefault()
```
Comment (argshook) [inline src/SectionHelper/SectionHelper.js]: @liorzisman guess you no longer need this?

File: src/SectionHelper/SectionHelper.scss [hunk 7]
```diff
@@ -18,6 +18,17 @@
   color: $D10;
 }
 
+.close {
+  position: absolute;
+  top: 10px;
+  right: 10px;
```
Comment (argshook) [inline src/SectionHelper/SectionHelper.scss]: @liorzisman please check if position will look okay when content has many rows

File: src/SectionHelper/SectionHelper.js (hunk 6 above)
Comment (liorzisman) [reply src/SectionHelper/SectionHelper.js]: Moved to the CloseButton default props where it was supposed to be :)

File: src/SectionHelper/SectionHelper.scss (hunk 7 above)
Comment (liorzisman) [reply src/SectionHelper/SectionHelper.scss]: looks great. i put more lines in the story content because its more of a realistic use case.

**Pull request merged**